pkg/core/mempool/mem_pool.go

```diff
@@ -609,3 +610,34 @@ func (mp *Pool) removeConflictsOf(tx *transaction.Transaction) {
 		}
 	}
 }
+
+// TryFindTransaction tries to find a transaction in the memory pool.
+// It searches through all the verified main and fallback transactions.
+func (mp *Pool) TryFindTransaction(hash util.Uint256) (*transaction.Transaction, bool) {
+	mp.lock.RLock()
+	defer mp.lock.RUnlock()
+
+	for i := range mp.verifiedTxes {
+		data := mp.verifiedTxes[i].data.(*payload.P2PNotaryRequest)
+		if data.MainTransaction.Hash() == hash {
+			return data.MainTransaction, true
+		} else if data.FallbackTransaction.Hash() == hash {
+			return data.FallbackTransaction, true
+		}
+	}
+	return nil, false
+}
```

Comment:
You don't need this method to retrieve both main and fallbacks. Use `TryGetValue(...)` to retrieve fallback tx by its hash. If this method doesn't return anything, then implement some predicate-based function `(*Mempool).TryFindData(isOk func(data any) bool) (any, bool)` to retrieve main tx by its hash. This method should work similar to `(mp *Pool) TryGetData(hash util.Uint256) (any, bool)` except that it will select the proper  `data` field using the given predicate and apply this predicate to each item in `mp.verifiedTxes` returning the first suitable value.

I'd like to isolate mempool from notary requests as much as possible and keep it generic as far as possible. @roman-khimov, what do you think about it?

Reply:
@tatiana-nspcc, do you have any problems with implementation?

Reply:
No, I just suffered while tried to implement JSON serialisation for util.Uint256. Done.